non-oss/pharos_pro/commands/license_assign_command.rb

```diff
@@ -27,8 +29,108 @@ def execute
 
       Dir.chdir(config_yaml.dirname) do
         master_host.transport.connect
-        master_host.transport.exec!("kubectl create secret generic pharos-cluster --namespace=kube-system --from-literal='license.jwt=#{jwt_token.token}' --dry-run -o yaml | kubectl apply -f -")
-        logger.info "Added subscription token to pharos cluster secrets"
+        master_host.transport.exec!("kubectl create secret generic pharos-license --namespace=kube-system --from-literal='license.jwt=#{jwt_token.token}' --dry-run -o yaml | kubectl apply -f -")
+        logger.info "Assigned the subscription token successfully to the cluster.".green
+      end
+    end
+
+    private
+
+    def default_license_key
+      if tty?
+        prompt.ask('Enter Kontena Pharos license key:')
+      elsif !stdin_eof?
+        $stdin.read
+      else
+        signal_usage_error 'LICENSE_KEY required'
+      end
+    end
+
+    def jwt_token
+      @jwt_token ||= if license_key.match?(/^\h{8}-(?:\h{4}-){3}\h{12}$/)
+                       Pharos::LicenseKey.new(subscription_token, cluster_id: cluster_id)
+                     else
+                       begin
+                         load_config
+                         c_id = cluster_id
+                       rescue StandardError
+                         c_id = nil
+                       end
+                       Pharos::LicenseKey.new(license_key, cluster_id: c_id)
+                     end
+    end
+
+    def master_host
+      @master_host ||= load_config.master_host
+    end
+
+    def cluster_info
```

Comment:
Maybe this should be in GatherFacts and store into cluster_context?

Anyway, as we now have gathered facts, we can do:
```
kube_client = load_config.kube_client(cluster_manager.context['kubeconfig'])
kube_client.api('v1').resource('configmaps', namespace: 'kube-public').get('cluster-info')&.metadata&.uid
```

It's now manipulated in the pro `MigrateMaster` phase.

Reply:
`cluster-info` is read using `kube_client` now